### ISSUE TITLE
Socket bind exception not captured in stdout log

### ISSUE THREAD

**Issue by mikeharder** [aspnet/IISIntegration#1125]
If two apps fail to start due to a port collision (see https://github.com/aspnet/IISIntegration/issues/1124), I would expect to see the managed exception (shown below) in the stdout log for the app.  However, for some reason this message is not captured in the stdout log.  I have verified a normal exception thrown from `Main()` is captured, and I've also verified that any calls to `Console.WriteLine()` before the port collision are also captured.

I have verified this impacts both v1 and v2 out-of-process.

The event log does contain messages from IIS which make it clear the issue is port collision, so this issue is a lower priority, but it's puzzling.

```
$ dotnet run
crit: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to start Kestrel.
System.IO.IOException: Failed to bind to address http://127.0.0.1:5000: address already in use. ---> Microsoft.AspNetCore.Connections.AddressInUseException: Only one usage of each socket address (protocol/network address/port) is normally permitted ---> System.Net.Sockets.SocketException: Only one usage of each socket address (protocol/network address/port) is normally permitted
   at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, String callerName)
   at System.Net.Sockets.Socket.DoBind(EndPoint endPointSnapshot, SocketAddress socketAddress)
   at System.Net.Sockets.Socket.Bind(EndPoint localEP)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransport.BindAsync()
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransport.BindAsync()
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServer.<>c__DisplayClass22_0`1.<<StartAsync>g__OnBind|0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context)
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.LocalhostListenOptions.BindAsync(AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.DefaultAddressStrategy.BindAsync(AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindAsync(IServerAddressesFeature addresses, KestrelServerOptions serverOptions, ILogger logger, Func`2 createBinding)
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServer.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken)
```

**Comment by mikeharder** [comment]
The word `crit` in the error message is colored red by default, which I thought might be the root cause.  However, the issue repros even with log colors disabled (https://github.com/aspnet/Logging/pull/772).

**Comment by pakrym** [comment]
The reason is that when ANCM sees that something is listening on the requested port but it's not a child process it short circuits and doesn't wait for the process or stdout logs.

Not sure if we want to fix this for 2.2 preview3 as it would be a medium sized change in out of proc.

**Comment by mikeharder** [comment]
> The event log does contain messages from IIS which make it clear the issue is port collision, so this issue is a lower priority

Closing since the event log contains sufficient information to diagnose port conflicts.  The stdout log would be nice to have, but it's not worth a big design change.
